test/UniswapV2Router01.spec.ts

```diff
@@ -16,546 +16,569 @@ const overrides = {
+        await WETHPair.mint(wallet.address, overrides)
+
+        const expectedLiquidity = expandTo18Decimals(2)
+
+        const nonce = await WETHPair.nonces(wallet.address)
+        const digest = await getApprovalDigest(
+          WETHPair,
+          { owner: wallet.address, spender: router.address, value: expectedLiquidity.sub(MINIMUM_LIQUIDITY) },
+          nonce,
+          MaxUint256
+        )
+
+        const { v, r, s } = ecsign(Buffer.from(digest.slice(2), 'hex'), Buffer.from(wallet.privateKey.slice(2), 'hex'))
+
+        await router.removeLiquidityETHWithPermit(
+          WETHPartner.address,
+          expectedLiquidity.sub(MINIMUM_LIQUIDITY),
+          0,
+          0,
+          wallet.address,
+          MaxUint256,
+          false,
+          v,
+          r,
+          s,
+          overrides
+        )
+      })
+
+      it('swapExactTokensForTokens', async () => {
+        const token0Amount = expandTo18Decimals(5)
+        const token1Amount = expandTo18Decimals(10)
+        await addLiquidity(token0Amount, token1Amount)
+
+        const swapAmount = expandTo18Decimals(1)
+        const expectedOutputAmount = bigNumberify('1662497915624478906')
+        await token0.approve(router.address, MaxUint256)
+        await expect(
+          router.swapExactTokensForTokens(
+            swapAmount,
+            0,
+            [token0.address, token1.address],
+            wallet.address,
+            MaxUint256,
+            overrides
+          )
+        )
+          .to.emit(token0, 'Transfer')
+          .withArgs(wallet.address, pair.address, swapAmount)
+          .to.emit(token1, 'Transfer')
+          .withArgs(pair.address, wallet.address, expectedOutputAmount)
+          .to.emit(pair, 'Sync')
+          .withArgs(token0Amount.add(swapAmount), token1Amount.sub(expectedOutputAmount))
+          .to.emit(pair, 'Swap')
+          .withArgs(router.address, swapAmount, 0, 0, expectedOutputAmount, wallet.address)
+      })
+
+      it('swapExactTokensForTokens:gas', async () => {
+        const token0Amount = expandTo18Decimals(5)
+        const token1Amount = expandTo18Decimals(10)
+        await addLiquidity(token0Amount, token1Amount)
+
+        // ensure that setting price{0,1}CumulativeLast for the first time doesn't affect our gas math
+        await mineBlock(provider, (await provider.getBlock('latest')).timestamp + 1)
+        await pair.sync(overrides)
+
+        const swapAmount = expandTo18Decimals(1)
+        await token0.approve(router.address, MaxUint256)
+        await mineBlock(provider, (await provider.getBlock('latest')).timestamp + 1)
+        const tx = await router.swapExactTokensForTokens(
+          swapAmount,
+          0,
+          [token0.address, token1.address],
+          wallet.address,
+          MaxUint256,
+          overrides
+        )
+        const receipt = await tx.wait()
+        expect(receipt.gasUsed).to.eq(routerVersion === RouterVersion.UniswapV2Router01 ? 101876 : 101876)
```

Comment:
is this just a placeholder for the next pr?

Reply:
yes, you'll see it's reworked in a later PR